src/datatype/shell.rs

```diff
@@ -10,3 +11,49 @@ pub fn system_info(cmd: &str) -> Result<String, Error> {
         .map_err(|err| Error::SystemInfo(err.to_string()))
         .and_then(|info| String::from_utf8(info.stdout).map_err(Error::FromUtf8))
 }
+
+
+/// Details of a remote OsTree package to pull.
+#[derive(RustcDecodable, Debug)]
+#[allow(non_snake_case)]
+pub struct OstreePackage {
+    pub commit:      String,
+    pub refName:     String,
+    pub description: String,
+    pub pullUri:     String
+}
+
+
+/// Install the specified `OstreePackage`.
+pub fn ostree_install(pkg: OstreePackage, token: Option<&AccessToken>) -> Result<String, Error> {
+    let mut command = Command::new("sota_ostree.sh");
+    command.env("COMMIT", pkg.commit)
+        .env("REF_NAME", pkg.refName)
+        .env("DESCRIPTION", pkg.description)
+        .env("PULL_URI", pkg.pullUri);
+    token.map(|t| command.env("AUTHPLUS_ACCESS_TOKEN", t.access_token.clone()));
+
+    command.output()
+        .map_err(|err| Error::OstreeCommand(err.to_string()))
+        .and_then(|ok| String::from_utf8(ok.stdout).map_err(Error::FromUtf8))
+}
+
+
+/// Return a list of installed ostree packages from `/usr/package.manifest`.
+pub fn ostree_installed_packages() -> Result<Vec<Package>, Error> {
+    Command::new("cat")
```

Comment:
None. It was implemented by someone else previously and I've just kept it as is for now.